### PR TITLE
KEYCLOAK-14235 Support for running broker tests with different hostnames for auth server and IdP

### PR DESCRIPTION
**Pull request by vmuzikar**
Since this PR slightly touches adapter tests as well, I run the tests here: https://keycloak-jenkins.com/job/universal-test-pipeline-adapters/1440/

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java [hunk 1]
```diff
@@ -286,14 +288,19 @@ public void testLinkAccountByLogInAsUserUsingBrowserButtons() {
...
         // Click browser 'back' on review profile page
         idpConfirmLinkPage.clickReviewProfile();
         waitForPage(driver, "update account information", false);
         updateAccountInformationPage.assertCurrent();
         driver.navigate().back();
+        // JS-capable browsers (i.e. all except HtmlUnit) add a new execution ID to the URL which then causes the login expire page to appear (because the old ID and new ID don't match)
+        if (!(driver instanceof HtmlUnitDriver)) {
+            loginExpiredPage.assertCurrent();
+            loginExpiredPage.clickLoginContinueLink();
+        }
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java:303]: I'm not sure if this is intended behavior. @mposolda Could you please double check this?

Comment (mposolda) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java:303]: @vmuzikar Yes, this is expected behaviour. IMO this change is fine

Comment (vmuzikar) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java:303]: Thanks!

**Comment by vmuzikar** [comment]
Rebased to resolve conflicts.

File: testsuite/integration-arquillian/HOW-TO-RUN.md [hunk 2]
```diff
@@ -925,3 +925,19 @@ Then, just run tests using the `auth-server-quarkus` profile:
...
 
+## Cookies testing
+In order to reproduce some specific cookies behaviour in browsers (like SameSite policies or 3rd party cookie blocking),
+some subset of tests needs to be ran with different hosts for auth server and app/IdP server in order to simulate third
+party contexts. Those hosts must be different from localhost as that host has some special treatment from browsers. At
+the same time both hosts must use different domains to be considered cross-origin, e.g. `127.0.0.1.nip.io` and
+`127.0.0.1.xip.io`. NOT `app1.127.0.0.1.nip.io` and `app2.127.0.0.1.nip.io`!!
+
+Also, those new cookies policies are currently not yet enabled by default (which will change in the near future). To test
+those policies, you need the latest stable Firefox and use `firefox-strict-cookies` profile.
```
Comment (mhajas) [inline testsuite/integration-arquillian/HOW-TO-RUN.md:936]: By default with this profile, firefox is running in headless mode. Could you please add some info about `firefoxHeadless` property here?

**Review by mhajas — COMMENTED** [review]
Great work @vmuzikar, thanks! I added just one minor comment about `firefoxHeadless` property.

File: testsuite/integration-arquillian/HOW-TO-RUN.md (hunk 2 above)
Comment (vmuzikar) [reply testsuite/integration-arquillian/HOW-TO-RUN.md]: Thanks for the review! I've updated the docs.

**Review by mhajas — DISMISSED** [review]
Thanks @vmuzikar!

**Comment by vmuzikar** [comment]
@mhajas Sorry for dismissing your review. Had to fix one stupid mistake :)

Before: https://github.com/keycloak/keycloak/blob/ff5b4875dbbae8a22415eadc3ab98446839f9b67/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/BrokerTestTools.java#L38

After: https://github.com/keycloak/keycloak/blob/f96929efa116cf63b39c2e0f2fdab87a4e99673d/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/BrokerTestTools.java#L39

**Comment by abstractj** [comment]
@vmuzikar once it gets approved by @mhajas or @mposolda, we can merge.

**Review by mhajas — DISMISSED** [review]
@vmuzikar Sorry I missed that. Approving again

**Comment by abstractj** [comment]
@vmuzikar we need a new rebase.

@mposolda assigned to you taking into consideration the changes in this PR. But please let me know if you won't have the time.

**Review by mposolda — APPROVED** [review]
@abstractj I've did just brief look, but looks fine. Relying on the @mhajas review mostly.

**Review by mposolda — CHANGES_REQUESTED** [review]
@vmuzikar Just the conflict in HOW-TO-RUN.md will need to be solved, which is quite minor thing.

**Comment by vmuzikar** [comment]
@mposolda Thanks for the review! Rebased.

**Review by abstractj — APPROVED** [review]

**Review by mposolda — APPROVED** [review]
@vmuzikar Thanks for the rebase!

**Comment by zak905** [comment]
Hello guys, seems like the changes introduced by this PR break the build, here to be more specific: https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/UsernameTemplateMapperTest.java

Here is an example from my latest PR build: https://github.com/keycloak/keycloak/runs/781246255

@vmuzikar would you mind taking a look please ?

**Comment by vmuzikar** [comment]
@zak905 Hi, thanks for reporting this. It was actually caused by #6927 and is fixed by #7178.

**Comment by zak905** [comment]
:+1:

**Pull request merged**